### ISSUE TITLE
InvalidDataAccessApiUsageException when do count on springboot

### ISSUE THREAD

**Issue by Nirvage** [bwaldvogel/mongo-java-server#66]
Hello, 

In my spring project, I migrated from fongo for mongo-java-server. 

But now, I encounter a problem with my count queries.

My requests look like this: 
```
    @CountQuery(value = "{'value.$id':'?0', 'object.value1':'?1', 'object.value2':'?2'}")
    int countItemsAffectedByValueAndValue1Value2(String value, String value1, String value2);
```

In my unit tests, the requests throw an error 59:

`org.springframework.dao.InvalidDataAccessApiUsageException: Command failed with error 59 (CommandNotFound): 'no such command: '$query'' on server 127.0.0.1:50778. The full response is { "$err" : "no such command: '$query'", "errmsg" : "no such command: '$query'", "code" : 59, "codeName" : "CommandNotFound", "bad cmd" : { "$query" : { "count" : "items", "query" : { "value.$id" : "any-id", "object.value1" : "FR", "object.value2" : "en" } }, "$readPreference" : { "mode" : "primaryPreferred" } }, "ok" : 0 }; nested exception is com.mongodb.MongoCommandException: Command failed with error 59 (CommandNotFound): 'no such command: '$query'' on server 127.0.0.1:50778. The full response is { "$err" : "no such command: '$query'", "errmsg" : "no such command: '$query'", "code" : 59, "codeName" : "CommandNotFound", "bad cmd" : { "$query" : { "count" : "products", "query" : { "value.$id" : "any-id", "objet.value1" : "FR", "objet.value2" : "en" } }, "$readPreference" : { "mode" : "primaryPreferred" } }, "ok" : 0 }`

My test configuration file looks like this:

```
    @Bean(destroyMethod = "shutdown")
    public MongoServer mongoServer() {
        MongoServer mongoServer = new MongoServer(new MemoryBackend());
        mongoServer.bind();
        return mongoServer;
    }

    @Bean(destroyMethod = "close")
    public MongoClient mongoClient(MongoServer mongoServer) {
        return new MongoClient(new ServerAddress(mongoServer.getLocalAddress()));
    }

    @Bean
    @Qualifier("idpMongoCollection")
    public MongoCollection<Document> idpMongoCollection() {
        return mongoClient(mongoServer()).getDatabase("idp_repository").getCollection("idp_repository");
    }
```

To be honest, i'm not sure if it's an issue or if I missed something

Do you have an idea about what’s wrong with my tests ?

I remain at your disposal for any further information.

Regards

**Comment by bwaldvogel** [comment]
I’m not yet able to reproduce it.
Which version of `mongo-java-server` and `spring-data` are you using?

**Comment by Nirvage** [comment]
Thanks for your anwser

I use the version 1.14.0 of mongo-java-server and my version of spring-data-jpa is 2.15.RELEASE

**Comment by bwaldvogel** [comment]
Which version of `mongo-java-driver` are you using?

**Comment by Nirvage** [comment]
I use the version 3.5.0, sorry for the response time

**Comment by bwaldvogel** [comment]
Could you create a minimized and reproducible test case?

**Comment by Nirvage** [comment]
Okay, I'll try to do that if I have availability this afternoon

**Comment by bwaldvogel** [comment]
This is probably fixed by #78 (version `1.17.1`)
